### PR TITLE
feat: adds new gestures to sil_euro_latin

### LINKED ISSUE #1975: [sil_euro_latin] feat: having common punctuation only under a long-press is annoying
From a team review of the Keyman for Android UX (keymanapp/keyman#7161)

A number of punctuation characters are available only under a long-press of the <kbd>.</kbd> key. While this is handy in some scenarios, we should consider if they should really also be available in the numeric/punctuation layer, and rarer punctuation characters moved to another layer or onto long-presses themselves.

### PR DESCRIPTION
**Pull request by jahorton**
This keyboard is designed to leverage many of the new gesture features landing in Keyman 17.0 for touch devices.  Until it releases, you'll need an up-to-date alpha or beta version of Keyman to test these out interactively.

For a pre-built version that doesn't require build-agent access: https://jahorton.github.io/sil_euro_latin.kmp

The new default layer:

touch-default

The 'shift' and 'caps' layers have the same hint-text (and down-flicks; more on that later).

The new numeric layer:
touch-numeric

The numeric layer, in particular, received a major layout change.  Note that each numeric base key matches the hint for its corresponding default-layer key.

For any 'hint' text on a text-output key (that isn't the old longpress-indicator), the _first_ letter can be output via down-flick on that key.  (On the parentheses keys, that means the square brackets are output by flick, but not the curly brackets.  Both are available as longpresses, though.)

The numeric, symbolic, and currency layer keys all accept multitap to rotate among them.

Fixes #1975.

### CODE REVIEW

**Comment by mcdurdin** [comment]
Note: this should not be merged until 17.0 is released.

**Comment by jahorton** [comment]
> Am I missing something? I see a version number change from 2.0.1 to 2.0.3 in the .kpj file, but I'd expect to see something similar in the .kmn file. Or is that coming later?

Later.  This is still in a "prototyping" / "drafting" phase.

Interesting re: change in the .kpj file; I didn't (knowingly) make any edits there.

File: release/sil/sil_euro_latin/source/welcome/touch-currency.png
Comment (jahorton) [inline release/sil/sil_euro_latin/source/welcome/touch-currency.png:1]: I used https://github.com/keymanapp/keyman/pull/10432?notification_referrer_id=NT_kwDOAYC52rM5MTIzNDM0NDg1OjI1MjEzNDAy#issuecomment-1897770448 to generate the new images.  It shouldn't take much effort to redo these if needed... so long as the resolution isn't set too high, anyway.

**Comment by mcdurdin** [comment]
We plan to test this with v16 now to make sure it still works there -- without the additional gestures of course -- so that we can merge this in early in 17.0 beta and get the benefit for all beta users.

**Comment by jahorton** [comment]
I've tested it out on Keyman 16.0 stable, and things are working fine there.  No gestures, of course, but the changes in layout are fully functional.

**Comment by mcdurdin** [comment]
Noted that this fixes #1975.

Could we add a small tweak to also fix #1976?

**Comment by jahorton** [comment]
> Noted that this fixes #1975.
> 
> Could we add a small tweak to also fix #1976?

I don't see why not; the trick lies in developing rules for it.  Any suggestions?  Alternatively, if you don't mind waiting, I could take a crack at it later in `beta`.

**Comment by darcywong00** [comment]
Does this mean the base branch changes to `master`?

**Comment by jahorton** [comment]
Do we need the 17.0 compiler to have the new gestures actually available for use?

**Comment by mcdurdin** [comment]
> Do we need the 17.0 compiler to have the new gestures actually available for use?

As far as I can remember, the gestures should compile just fine with the 16.0 compiler.

**Comment by mcdurdin** [comment]
I'm going to have to update this to get it based on master rather than staging-17.0 (or @jahorton can do this).

There will also need to be metadata updates: 

> > Am I missing something? I see a version number change from 2.0.1 to 2.0.3 in the .kpj file, but I'd expect to see something similar in the .kmn file. Or is that coming later?
> 
> Later. This is still in a "prototyping" / "drafting" phase.

This needs to be done before review.

> Interesting re: change in the .kpj file; I didn't (knowingly) make any edits there.

The metadata in the .kpj file should be ignored -- the metadata is generated by the IDE when changes are made to other files, so it can get out of sync.

**Comment by mcdurdin** [comment]
Metadata updated. .kmn has 3.0.0, .kpj does too (though can be ignored), HISTORY.md includes details of changes, LICENSE.md years updated.

**Review by LornaSIL — APPROVED** [review]
LGTM

**Pull request merged**